### PR TITLE
completed assessment

### PR DESCRIPTION
**Pull request by HackCoder77**
FUN CODING!!!!

### CODE REVIEW

File: code-challenges.js [hunk 1]
```diff
@@ -9,40 +9,81 @@
 
 // Create the code and test each of the variables provided. Expected output: "35 is below boiling point", "350 is above boiling point", "212 is at boiling point"
 
+//I need to use 'if' and 'else if' statements to set a range for the numbers to output desired. I will have to set 3 outputs based around 212 . One at temp, one below and one above. I will use template literals to console back the exact temperature that was entered.
```
Comment (WalkerTexas-coder) [inline code-challenges.js:12]: well done! I remember you mentioning struggling with pseudo code and this is great!

File: code-challenges.js [hunk 2]
```diff
@@ -9,40 +9,81 @@
...
+//At first i tried using the 'reverse' command but i forgot that was an Array method. Now i have to look up reversing strings (slipped my mind how). So, i need to use 'split' first which returns an array, then i can continue with what i was going to do with 'reverse' and 'join'.  I will need to create a function to complete this. Using string as the argument. I will then run the function for both strings.
+ 
 var myString1 = "charlie"
 var myString2 = "delta"
 
+function reverse (string){
+    return console.log(string.split("").reverse().join(""))
+}
+reverse(myString1);
+reverse(myString2);
```
Comment (WalkerTexas-coder) [inline code-challenges.js:51]: nicely done

File: code-challenges.js [hunk 3]
```diff
@@ -9,40 +9,81 @@
...
+
 var myNumbers1 = [3, 7, 0, 36, -9]
 var myNumbers2 = [8, -7, 22, 9, 13]
 
+myNumbers1.concat(myNumbers2).length
 
 
 // --------------------3) Create the code that will reverse the letters of a string using the test variables provided below. Expected output: "eilrahc", "atled"
 
+//At first i tried using the 'reverse' command but i forgot that was an Array method. Now i have to look up reversing strings (slipped my mind how). So, i need to use 'split' first which returns an array, then i can continue with what i was going to do with 'reverse' and 'join'.  I will need to create a function to complete this. Using string as the argument. I will then run the function for both strings.
```
Comment (WalkerTexas-coder) [inline code-challenges.js:42]: great observations!

File: code-challenges.js [hunk 4]
```diff
@@ -9,40 +9,81 @@
...
+} else {
+    console.log(number2 - number1)
+
+}
 
 
 // --------------------5) Copy/paste your code from #4. Refactor your code to also accept non-number edge cases. Inform your user if the variable is not a number using the two sets of test variables provided below. Expected output: "Your input is not a number", 3
 
+//In this case now i need to use 'typeof' to be able to make sure that the only type of information i take in is numbers. I will write it so that it will say if this input is NOT a number, then console.log warning message. I will have to also use 'or' since there is a second variable. If they are both numbers then i need it to go to 'else' statements to finish it off.
+
```
Comment (WalkerTexas-coder) [inline code-challenges.js:75]: Nice!

File: coding-concepts.js [hunk 5]
```diff
@@ -8,44 +8,45 @@
...
 
 var weekendDays = ["saturday", "sunday"]
-// console.log(weekendDays.toUpperCase())
+console.log(weekendDays.toUpperCase())
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: "SATURDAY", "SUNDAY"
+// b) Verify and explain:toUpperCase() is a built in function that changes lowercase letters to all uppercase.
+// Correct answer was that toUpperCase is not a function. Trick question got me!
```
Comment (WalkerTexas-coder) [inline coding-concepts.js:43]: Glad it worked! Yeah, keeping track of the data type string vs array and knowing which methods work on each is part of the job.

File: coding-concepts.js [hunk 6]
```diff
@@ -8,44 +8,45 @@
...
 // --------------------5) What will this log?
 
 var dataTypes = ["number", "string", "Boolean"]
-// console.log(typeof dataTypes.length)
+console.log(typeof dataTypes.length)
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: number
+// b) Verify and explain: because it asked for length and length is a number so we get number.
```
Comment (WalkerTexas-coder) [inline coding-concepts.js:52]: perfect!

File: interview-questions.md [hunk 7]
```diff
@@ -7,60 +7,63 @@ Then, research the question to expand on your answer. Even if you feel you have
...
-  Researched answer:
+  Researched answer: 
+  Relational, logical (&&, ||, !), equality
 
 
 
 3. What is an index? What is the difference between index and value?
 
-  Your answer:
+  Your answer: Index is the location and the difference is Index is a location for where a value is and value is the subject.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:29]: Well said

File: interview-questions.md [hunk 8]
```diff
@@ -7,60 +7,63 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+  Researched answer: Index is the location. Value is the meaning 
 
 
 
 4. What is a built-in method?
 
-  Your answer:
+  Your answer: a function already designed into the JS language that runs by us just typing a short command.
 
   Researched answer:
```
Comment (WalkerTexas-coder) [inline interview-questions.md]: Just a heads up these answers are required for a complete assessment.

**Review by WalkerTexas-coder — CHANGES_REQUESTED** [review]
Ato, most everything here is great! I am sending it back to you so you can push up a version with all the interview answers. Please remember answers should be articulated like you would in an actual interview.

File: interview-questions.md [hunk 9]
```diff
@@ -7,60 +7,63 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
-  Researched answer:
+  Researched answer: Index is the location. Value is the meaning 
 
 
 
 4. What is a built-in method?
 
-  Your answer:
+  Your answer: a function already designed into the JS language that runs by us just typing a short command.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:37]: yes!

**Review by WalkerTexas-coder — APPROVED** [review]
Ato, solid work! Your code is awesome and I really appreciate how much effort you put into the pseudo code! My encouragement for moving forward is spend a bit more time on the interview questions. They're going to be a great resource for you to look back on when you head to actual interviews. Maybe focus on "the correct answer" and then explaining what that would look like in your experience. That really helps you connect with an interviewer. Keep up the good work! Your PR is approved!
